test/e2e/kubernetes/kubernetes.go

```diff
@@ -40,27 +40,28 @@ const (
 	MinimumVersion           = "1.25"
 )
 
-// WaitForNode wait for the node to join the cluster and fetches the node info from an internal IP address of the node
-func WaitForNode(ctx context.Context, k8s kubernetes.Interface, internalIP string, logger logr.Logger) (*corev1.Node, error) {
+// WaitForNode wait for the node to join the cluster and fetches the node info which has the nodeName label
+// will continue to poll until the node has an internal IP address
+func WaitForNode(ctx context.Context, k8s kubernetes.Interface, nodeName string, logger logr.Logger) (*corev1.Node, error) {
 	foundNode := &corev1.Node{}
 	consecutiveErrors := 0
 	err := wait.PollUntilContextTimeout(ctx, hybridNodeDelayInterval, hybridNodeWaitTimeout, true, func(ctx context.Context) (done bool, err error) {
-		node, err := getNodeByInternalIP(ctx, k8s, internalIP)
+		node, err := getNodeByE2ELabelName(ctx, k8s, nodeName)
 		if err != nil {
 			consecutiveErrors += 1
 			if consecutiveErrors > 3 {
 				return false, err
 			}
-			logger.Info("Retryable error listing nodes when looking for node with IP. Continuing to poll", "internalIP", internalIP, "error", err)
+			logger.Info("Retryable error listing nodes when looking for node with name. Continuing to poll", "nodeName", nodeName, "error", err)
 			return false, nil // continue polling
 		}
 		consecutiveErrors = 0
-		if node != nil {
+		if node != nil && GetNodeInternalIP(node) != "" {
```

Comment:
should this be in `WaitForNodeReady`?
I feel like this method should just wait until a matching node exists, not until a matching node has an IP